### ISSUE TITLE
Humanoid Avatar Rig Not Working

### ISSUE THREAD

**Issue by JeanxPereira** [KhronosGroup/UnityGLTF#796]
### Describe the bug 💬

I can't apply the modification to the Humanoid character and create the Avatar element for it. It says that LeftUpperLeg is missing from the model, even though it is correctly configured on both the character and in AvatarUtils.

### **`Error:`**
![image](https://github.com/user-attachments/assets/06472813-6a69-4500-b551-3d987e3d3b54)

### **`AvatarUtils:`**
![image](https://github.com/user-attachments/assets/53dda7de-3bcb-4eee-aa6c-0e728dd2b4b2)

### **`GLTF containing L_Hip, which is interpreted as LeftUpperLeg in AvatarUtils:`**
![image](https://github.com/user-attachments/assets/25aacafc-724d-4b27-bee6-dbbe325a2545)

### Steps to reproduce 🔢

1. Import the GLTF model into Unity.
2. Verify that the bone names are correctly mapped in AvatarUtils.
3. Create the Avatar using the Avatar menu in the prefab generated by the GLTF Plugin.
4. Error: Missing LeftUpperLeg.

### Files to reproduce the issue ♻

https://drive.google.com/drive/folders/1d72fK9oWq1OY7NH4RuqtHdd9p2FA4yz7?usp=sharing

### Editor Version 🎲

6000.0

### Render Pipeline and version

Built-In

### UnityGLTF Version

2.14.1

### Operating System 👩‍💻

Windows

### When does this problem happen?

- [X] Editor Import
- [X] Runtime Import
- [ ] Editor Export
- [ ] Runtime Export

### Additional Info 📜

This error also occurs for other people using different plugins, such as GLTFUtility, where I have also reported this issue:
https://github.com/Siccity/GLTFUtility/issues/70

### Validations 🩹

- [X] I have [searched existing issues](https://github.com/KhronosGroup/UnityGLTF/issues): no issue already exist that reports the same problem.
- [X] I follow the [Code of Conduct](https://www.khronos.org/about/code-of-conduct)
- [X] I provided a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), including files when necessary.

**Comment by hybridherbst** [comment]
Hey, the name of the node seems to be "L_Hip" but your mapping looks for "riotleague:L_Hip" which is a different string.
Can you try adding the actual name of the node without that prefix?

**Comment by JeanxPereira** [comment]
hey, I also did this test by modifying the bones directly in the GLTF file using the names used in Unity, but it still didn’t work

**Comment by hybridherbst** [comment]
Can you upload those files and what error message you get as well please? Thanks

**Comment by JeanxPereira** [comment]
yes, are the files I sent mentioned in:

### Files to reproduce the issue ♻

https://drive.google.com/drive/folders/1d72fK9oWq1OY7NH4RuqtHdd9p2FA4yz7?usp=sharing

**Comment by XtroTheArctic** [comment]
Was there any resolution for this in the last 6-7 months?

**Comment by XtroTheArctic** [comment]
I recently started using UnityGLTF and I am not experienced in avatar creation but I'll try looking into this. Let's see...

**Comment by JeanxPereira** [comment]
cool, if possible keep updates here 🙂

**Comment by XtroTheArctic** [comment]
@JeanxPereira I investigated your usage case by importing gltf files and AvatarUtils you provided.

First of all, when I try to import darkstar_thresh I get this error: "AvatarBuilder 'SkinnedMesh': Required human bone 'Hips' not found"; not the error you first reported "leftupperleg".

Secondly, you (we, all unity developers) are not supposed to modify code files in 3rd party packages. They are supposed to remain as read-only. Even Unity shows a warning about this.

Moreover, you injected you own "riotleague" strings to the dictionary but:
1. You model doesn't contain bone names starting with "riotleague" as shown in the screenshot below.
2. The code using this dictionary is not looking for exact bone names. It's using the dictionary to rename some recognized names into different names Unity can understand such as renaming "Spine1" to "Chest". Some bones in your model has regular names like Neck and Head so it is able to recognize them but the bones like L_Hip aren't recognized because there is no L_Hip entry in the dictionary.
3. There is a hard coded piece of code trying to add "mixamorig" prefix to bone names in an attempt to recognize them but it's still not able to recognize them because you commented out those mixamorig dictionary entries.
4. I manually changed that piece of code from "mixamorig" to "riotleague" on line 302 but it still doesn't work because it's looking for lowercase names and the riotleague entries you added are not in all lowercase.
5. I prevented the code to use lowercase when looking for the bone names starting with riotleague and now it's able to find the hip bone
6. But this time it can't find the spine bone because there is no "Spine" bone in your model. There are Spine1 and Spine2 but they are used for Chest.

So, as a result, your setup has multiple problems and this ticket can't be considered as a UnityGLTF bug, especially when you modify a read-only code file.

I, myself, am very much aware that there are bone name recognition problems both in UnityGTLF and Unity itself but this is not one of them.

Now I'll go back to another ticket reported by another person (which I experience the same problem too). If I can figure it out and apply some kind of solution, I'll submit a PR and hopefully solve these bone detection issues.
https://github.com/KhronosGroup/UnityGLTF/issues/794

@JeanxPereira I can suggest you to rename the bone names in your model inside the modelling application you are using.

I think this ticket can be closed. @hybridherbst 

![Image](https://github.com/user-attachments/assets/502dcb0b-a273-4e0c-8a4d-8b86fef24498)

**Comment by JeanxPereira** [comment]
wow, very good
yea, I modified the AvatarUtils file after failing other attempts to import thresh into the project but even without modifying anything and using the correct names in gltf the error continued

**Comment by XtroTheArctic** [comment]
The thing is... even Unity's FBX importer can't support many character models as is. BUT, it provides an avatar configuration system where we can assign the bones manually. That's how we were able to use many models correctly for many years. Unfortunately UnityGLTF doesn't provide that avatar configuration thing so, yea, many models will not work with GLTF unfortunately.

Like I said, I am not experienced in this area but I'll see what I can do in the other ticket because I need the solution too.

**Comment by JeanxPereira** [comment]
what a shame :(
Thanks again, what is the ticket so I can follow up? I'll close this issue

**Comment by hybridherbst** [comment]
> Unfortunately UnityGLTF doesn't provide that avatar configuration thing so, yea
Yeah, that's why currently we need repro files to check common avatar configurations, so that we can support common ones. Unity does not expose their avatar rig setup tools and recreating them from scratch is out of scope for this project.
